src/app/modules/shared_models/payment_request_model.nim

```diff
@@ -0,0 +1,111 @@
+import NimQml, stew/shims/strformat, tables, sequtils
+import ../../../app_service/service/message/dto/payment_request
+
+type
+  ModelRole {.pure.} = enum
+    Symbol = UserRole + 1
+    Amount
+    ReceiverAddress
+    ChainId
+
+QtObject:
+  type
+    Model* = ref object of QAbstractListModel
+      items: seq[PaymentRequest]
+
+  proc delete*(self: Model) = 
+    self.items = @[]
+    self.QAbstractListModel.delete
+
+  proc setup(self: Model) =
+    self.QAbstractListModel.setup
+
+  proc newPaymentRequestModel*(paymentRequests: seq[PaymentRequest] = @[]): Model =
+    new(result, delete)
+    result.setup
+    result.items = paymentRequests
+
+  proc newModel*(): Model =
+    new(result, delete)
+    result.setup
+
+  proc `$`*(self: Model): string =
+    for i in 0 ..< self.items.len:
+      result &= fmt"""
+      [{i}]:({$self.items[i]})
+      """
+
+  proc getPaymentRequests*(self: Model): seq[PaymentRequest] =
+    return self.items
+
+  proc countChanged(self: Model) {.signal.}
```

Comment:
It's better to use the `ModelCount` attached property instead of the nim `count` signal.
In qml you can do something like this:

```
anyModel.ModelCount.count
```